src/icrar/leap-accelerate-cli/main.cc

```diff
@@ -46,55 +45,58 @@ namespace icrar
     enum class InputType
     {
         STREAM,
-        FILE_STREAM,
         FILENAME,
         APACHE_ARROW
     };
 
+    /**
+     * Raw set of command line interface arguments
+     */
     struct Arguments
     {
         InputType source = InputType::FILENAME;
-        boost::optional<std::string> filePath;
-        boost::optional<std::string> directions;
+        boost::optional<std::string> filePath = boost::none; // Measurement set filepath
+        boost::optional<std::string> configPath = boost::none; // Config filepath
+
+        boost::optional<std::string> stations = boost::none;
+        boost::optional<std::string> directions = boost::none;
         ComputeImplementation implementation = ComputeImplementation::casa;
+
+        Arguments() {}
     };
 
+    /**
+     * Validated set of command line arguments
+     */
     class ArgumentsValidated
     {
-        InputType m_source;
-        boost::optional<std::string> m_filePath;
+        /**
+         * Constants
+         */
+        InputType m_source; // MeasurementSet source type
+        boost::optional<std::string> m_filePath; // MeasurementSet filepath
+        boost::optional<std::string> m_stations; // Overriden number of stations
         std::vector<MVDirection> m_directions;
         ComputeImplementation m_computeImplementation;
 
         /**
          * Resources
          */
         std::unique_ptr<MeasurementSet> m_measurementSet;
-        std::ifstream m_fileStream;
         std::istream* m_inputStream = nullptr; // Cached reference to the input stream
 
     public:
         ArgumentsValidated(const Arguments&& args)
```

Comment:
If you choose to receive by rvalue reference then you probably want to move things out of `args` instead of copying, but this isn't happening: While `args` itself is constructed by rvalue ref, it is marked as `const` and then its members are copied rather than moved (e.g., `m_source(args.sources)` rather than `m_source(std::move(args.source))`)